packages/osd-stylelint-plugin-stylelint/src/rules/invalid_properties/index.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import stylelint from 'stylelint';
+import { NAMESPACE } from '../..';
+import {
+  getNotCompliantMessage,
+  getPropertyRule,
+  getRulesFromConfig,
+  isValidOptions,
+} from '../../utils';
+
+const { ruleMessages, report } = stylelint.utils;
+
+const ruleName = 'invalid_properties';
+const messages = ruleMessages(ruleName, {
+  expected: (message) => `${message}`,
+});
+
+const ruleFunction = (
+  primaryOption: Record<string, any>,
+  secondaryOptionObject: Record<string, any>,
+  context
+) => {
+  return (postcssRoot: any, postcssResult: any) => {
+    const validOptions = isValidOptions(postcssResult, ruleName, primaryOption);
+    if (!validOptions) {
+      return;
+    }
+
+    const rules = getRulesFromConfig(primaryOption.config);
+
+    const isAutoFixing = Boolean(context.fix);
+
+    postcssRoot.walkDecls((decl: any) => {
+      const propertyRule = getPropertyRule(rules, decl);
+      if (!propertyRule) {
+        return;
+      }
+
+      let shouldReport = false;
+
+      const file = postcssRoot.source.input.file;
+      const approvedFiles = propertyRule.approved;
+
+      const reportInfo = {
+        ruleName: `${NAMESPACE}/${ruleName}`,
+        result: postcssResult,
+        node: decl,
+        message: '',
+      };
+
+      if (approvedFiles) {
+        shouldReport = !approvedFiles.some((inspectedFile) => {
+          return file.includes(inspectedFile);
+        });
+      }
+
+      if (shouldReport && isAutoFixing) {
+        decl.remove();
```

Comment:
nice! I like it as it removes just the specific property instead of removing the entire style.

I wonder if we backlog an improvement to think about checking the styleblock after this auto fix and deleting the entire style node if empty.

For example,

```scss
.myTitle {
  font-family: 'Courier New'
}
```

will turn into

```scss
.myTitle {
}
```

Which then gets caught by: `block-no-empty` stylelint rule. 

But can be a follow-up,  if we think is needed. The behavior you have of just deleting the property is the correct method here. And I'm fine leaving it as is for now.

Reply:
I _feel_ like this could be easy, i.e. get parent class, check how many styles after removing, if zero remove class. Need to look into it though.